clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1504,6 +1507,8 @@ public void handleResponse(AbstractResponse response) {
                 fatalError(error.exception());
             } else if (error == Errors.GROUP_AUTHORIZATION_FAILED) {
                 abortableError(GroupAuthorizationException.forGroupId(key));
+            } else if (error == Errors.ABORTABLE_TRANSACTION_EXCEPTION) {
```

Comment:
Does FindCoordinatorRequest also return the ABORTABLE_TRANSACTION_EXCEPTION?

Reply:
The idea of this error is that we are future proofing. In the future if we want the producer to abort the transaction (say we have a new use case that requires this behavior) we can rely on this error to do the correct thing for older clients. 

We ran into this issue a lot when picking error codes for kip-890 part 1. If we had such an error then, we could have used it for the old clients. Instead, we chosen INVALID_TXN_STATE which has inconsistent behavior across clients.